lib/matplotlib/contour.py

```diff
@@ -1117,17 +1117,16 @@ def _autolev(self, N):
 
         return lev[i0:i1]
 
-    def _process_contour_level_args(self, args):
+    def _process_contour_level_args(self, args, z_dtype):
         """
         Determine the contour levels and store in self.levels.
         """
-        if self.levels is None:
-            if len(args) == 0:
-                levels_arg = 7  # Default, hard-wired.
-            else:
-                levels_arg = args[0]
-        else:
-            levels_arg = self.levels
+        levels_arg = (
+            self.levels if self.levels is not None else
+            args[0] if args else
+            # z is bool
+            ([0, .5, 1] if self.filled else [.5]) if z_dtype.char == "?" else
+            7)  # Hard-coded default.
```

Comment:
This is quite unreadable compared to a "standard" if-else. I see the point if there are two options, but this takes quite a bit of time to decode.

Reply:
Sure, I spelled it out.